src/RSocketClient.cpp

```diff
@@ -43,11 +44,21 @@ folly::Future<std::unique_ptr<RSocketRequester>> RSocketClient::connect(
     networkStats = std::move(networkStats),
     promise = std::move(promise)](
       std::unique_ptr<DuplexConnection> connection,
+      bool isFramedConnection,
```

Comment:
I would suggest, since this is already a large change, to generate a new diff with this particular change.
The idea for the diff would be to make FramedDuplexConnection an internal type which would not be used by the application code or implementors of DuplexConnection interface. Instead they will provide a boolean parameter to RSocketClient/RSocketServer whether the connection is framed or not (so far all implementations we have are not framed). Then internally we would do what the lines 51-59 are doing.

The main motivator for this change is to avoid application code to specify protocol version for FramedDuplexConnection.
cc: @somasun